src/commands/synthetics/run-tests-lib.ts

```diff
@@ -192,52 +168,117 @@ const getTestListBySearchQuery = async (
+    config: {
+      ...config.global,
+      ...configFromEnvironment,
+    },
+    id,
+  }))
 
-    if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
-      reporter.error(
-        `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
+  if (!testsToTrigger.length && config.testSearchQuery) {
+    try {
+      testsToTrigger = await getTestsFromSearchQuery(api, config, reporter)
+      testsToTrigger = testsToTrigger.map((test) => ({
+        ...test,
+        config: {
+          ...configFromEnvironment,
+          ...test.config,
+        },
+      }))
+    } catch (error) {
+      throw new CriticalError(
+        isForbiddenError(error) ? 'AUTHORIZATION_ERROR' : 'UNAVAILABLE_TEST_CONFIG',
+        error.message
       )
     }
+  }
 
-    return testsToTriggerBySearchQuery
+  try {
+    testConfig = await getTestConfig(config, reporter, suites)
+  } catch (error) {
+    throw new CriticalError(isForbiddenError(error) ? 'AUTHORIZATION_ERROR' : 'UNAVAILABLE_TEST_CONFIG', error.message)
+  }
+
+  if (!testsToTrigger.length) {
+    testsToTrigger = testConfig.map((test) => ({
+      ...test,
+      config: {
+        ...config.global,
+        ...configFromEnvironment,
+        ...test.config,
+      },
+    }))
+  } else {
+    testsToTrigger = overrideWithTestConfig(testsToTrigger, testConfig)
+  }
+
+  if (!testsToTrigger.length) {
+    throw new CiError('NO_TESTS_TO_RUN')
+  }
+
+  return testsToTrigger
+}
+
+const getTestsFromSearchQuery = async (api: APIHelper, config: RunTestsCommandConfig, reporter: MainReporter) => {
+  const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery || '')
+
+  if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
+    reporter.error(
+      `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
+    )
   }
 
+  return testsToTriggerBySearchQuery
+}
+
+const getTestConfig = async (
+  config: RunTestsCommandConfig,
+  reporter: MainReporter,
+  suites: Suite[] = []
+): Promise<TriggerConfig[]> => {
   const suitesFromFiles = (await Promise.all(config.files.map((glob: string) => getSuites(glob, reporter))))
     .reduce((acc, val) => acc.concat(val), [])
     .filter((suite) => !!suite.content.tests)
 
   suites.push(...suitesFromFiles)
 
-  const configFromEnvironment = config.locations?.length ? {locations: config.locations} : {}
-
-  const overrideTestConfig = (test: TriggerConfig): UserConfigOverride =>
-    // {} < global < ENV < test file
-    ({
-      ...config.global,
-      ...configFromEnvironment,
-      ...test.config,
-    })
-
-  const testsToTrigger = suites
+  const testsFromTestConfig = suites
     .map((suite) =>
       suite.content.tests.map((test) => ({
-        config: overrideTestConfig(test),
+        config: {...test.config},
```

Comment:
Do we need this spread here? Can't we just pass `test.config`?
```suggestion
        config: test.config,
```

Reply:
artefact from earlier when i was spreading something else in there as well 👍 good catch